RecoLocalCalo/HGCalRecAlgos/plugins/alpaka/HGCalRecHitProducers.cc

```diff
@@ -0,0 +1,164 @@
+    //                                   .erxChannelMax = 37+2,//+2 for the two common modes
+    // });
+  }
+
+  void HGCalRecHitsProducer::produce(device::Event& iEvent, device::EventSetup const& iSetup) {
+    auto queue = iEvent.queue();
+
+    // Read digis
+    auto const& deviceCalibParamProvider = iSetup.getData(calibToken_);
+    auto const& deviceConfigParamProvider = iSetup.getData(configToken_);
+    auto const& hostDigisIn = iEvent.get(digisToken_);
+
+// Check if there are new conditions and read them
+#ifdef EDM_ML_DEBUG
+    if (calibWatcher_.check(iSetup)) {
+      for (int i = 0; i < deviceConfigParamProvider.view().metadata().size(); i++) {
+        LogDebug("HGCalCalibrationParameter") << "gain = " << deviceConfigParamProvider.view()[i].gain();
+      }
+    }
+#endif
+
+// Check if there are new conditions and read them
+#ifdef EDM_ML_DEBUG
+    if (configWatcher_.check(iSetup)) {
+      for (int i = 0; i < deviceCalibParamProvider.view().metadata().size(); i++) {
+        LogDebug("HGCalCalibrationParameter")
+            << "idx = " << i << ", "
+            << "ADC_ped = " << deviceCalibParamProvider.view()[i].ADC_ped() << ", "
+            << "CM_slope = " << deviceCalibParamProvider.view()[i].CM_slope() << ", "
+            << "CM_ped = " << deviceCalibParamProvider.view()[i].CM_ped() << ", "
+            << "BXm1_slope = " << deviceCalibParamProvider.view()[i].BXm1_slope() << ", " << std::endl;
+      }
+    }
+#endif
+
+    int oldSize = hostDigisIn.view().metadata().size();
+    int newSize = oldSize * n_hits_scale;
+    auto hostDigis = HGCalDigiHost(newSize, queue);
+    // TODO: replace with memcp ?
+    for (int i = 0; i < newSize; i++) {
+      //hostDigis.view()[i].electronicsId() = hostDigisIn.view()[i%oldSize].electronicsId();
+      hostDigis.view()[i].tctp() = hostDigisIn.view()[i % oldSize].tctp();
+      hostDigis.view()[i].adcm1() = hostDigisIn.view()[i % oldSize].adcm1();
+      hostDigis.view()[i].adc() = hostDigisIn.view()[i % oldSize].adc();
+      hostDigis.view()[i].tot() = hostDigisIn.view()[i % oldSize].tot();
+      hostDigis.view()[i].toa() = hostDigisIn.view()[i % oldSize].toa();
+      hostDigis.view()[i].cm() = hostDigisIn.view()[i % oldSize].cm();
+      hostDigis.view()[i].flags() = hostDigisIn.view()[i % oldSize].flags();
+      //LogDebug("HGCalCalibrationParameter")
+      //  << "idx=" << i << ", elecId=" << hostDigis.view()[i].electronicsId()
+      //  << ", cm=" << hostDigis.view()[i].cm() << std::endl;
+    }
+    LogDebug("HGCalRecHitsProducer") << "Loaded host digis: " << hostDigis.view().metadata().size();  //<< std::endl;
+
+    LogDebug("HGCalRecHitsProducer") << "\n\nINFO -- calling calibrate method";  //<< std::endl;
+    auto start = now();
+    auto recHits = calibrator_.calibrate(queue, hostDigis, deviceCalibParamProvider, deviceConfigParamProvider);
+    alpaka::wait(queue);
+    auto stop = now();
```

Comment:
I think by default the time measurement and especially the `wait()` should be avoided.

In addition, in order for the time measurement to not include any prior work queued in the `queue`, it would be good to `wait()` before starting the clock.